apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx

```diff
@@ -3,17 +3,54 @@ import { useState, useEffect } from 'react'
 import {
   useListPublicKeyQuery,
   ListPublicKeyQueryQueryParams,
-  ListPublicKeyOkResponse
-  // ListPublicKeyErrorResponse
+  ListPublicKeyOkResponse,
+  useCreateTokenMutation,
+  CreateTokenOkResponse,
+  CreateTokenErrorResponse,
+  CreateTokenRequestBody,
+  useCreatePublicKeyMutation,
+  CreatePublicKeyRequestBody,
+  CreatePublicKeyOkResponse,
+  CreatePublicKeyErrorResponse,
+  ListPublicKeyErrorResponse
 } from '@harnessio/code-service-client'
-
+import { TokenCreateDialog } from './token-create/token-create-dialog'
+import { SshKeyCreateDialog } from './ssh-key-create/ssh-key-create-dialog'
+import { TokenSuccessDialog } from './token-create/token-success-dialog'
 import { TokensList } from '@harnessio/playground'
 
 export const SettingsProfileKeysPage = () => {
   const TEMP_USER_TOKENS_API_PATH = '/api/v1/user/tokens'
 
-  const [publicKeys, setPublicKeys] = useState<ListPublicKeyOkResponse[]>([])
+  const [publicKeys, setPublicKeys] = useState<ListPublicKeyOkResponse>([])
   const [tokens, setTokens] = useState<TokensList[]>([])
+  const [openCreateTokenDialog, setCreateTokenDialog] = useState(false)
+  const [openSuccessTokenDialog, setSuccessTokenDialog] = useState(false)
+  const [saveSshKeyDialog, setSshKeyDialog] = useState(false)
+  const [apiError, setApiError] = useState<{
+    type: 'keyFetch' | 'tokenFetch' | 'keyCreate' | 'tokenCreate'
+    message: string
+  } | null>(null)
+
+  const [createdTokenData, setCreatedTokenData] = useState<{
+    identifier: string
+    lifetime: string
+    token: string
```

Comment:
Can we use existing type here? https://github.com/harness/react-api-client/blob/main/packages/code-service/src/services/schemas/TypesTokenResponse.ts#L6

Reply:
We could, but it doesn't conform. We cherry pick fields we need for the UI. They're pretty different, not a good idea in my opinion.

```
  const tokenData = {
          identifier: newToken.token?.identifier ?? 'Unknown',
          lifetime: newToken.token?.expires_at
            ? new Date(newToken.token.expires_at).toLocaleDateString()
            : 'No Expiration',
          token: newToken.access_token ?? 'Token not available'
        }
```